lib/membrane_live/event_service.ex

```diff
@@ -44,6 +51,64 @@ defmodule MembraneLive.EventService do
   @impl true
   def init(_opts), do: {:ok, %__MODULE__{}}
 
+  @doc """
+  Adds peer to the given event's room
+  """
+  @spec add_peer(event_id(), peer_id()) :: :ok | room_error()
+  def add_peer(event_id, peer_id) do
+    case :global.whereis_name(event_id) do
+      :undefined ->
+        room_doesnt_exist_error()
+
+      pid ->
+        Room.add_peer(pid, peer_id)
+    end
+  end
+
+  @doc """
+  Removes peer from given event's room
+  """
+  @spec remove_peer(event_id(), peer_id()) :: :ok | room_error()
+  def remove_peer(event_id, peer_id) do
+    case :global.whereis_name(event_id) do
+      :undefined ->
+        room_doesnt_exist_error()
+
+      pid ->
+        Room.remove_peer(pid, peer_id)
+    end
+  end
```

Comment:
because those functions have different numbers of parameters I would have to write it using Lambdas 
```
def add_peer(event_id, peer_id) do
   add_peer = fn pid -> Room.add_peer(pid, peer_id) end
   proceed_if_room_exist(add_peer, event_id)
end
```
So I would prefer to stick to the current version. Or we could also use `with`:
```
defp get_room(event_id) do
   case :global.where_is(event_id) do
      :undefined ->
        room_doesnt_exist_error()

      pid ->
        {:ok, pid}
    end
end

def add_peer(event_id, peer_id) do
   with {:ok, pid} <- get_room(event_id) do
      Room.add_peer(pid, peer_id)
   end
end
```
What do you think?

Reply:
Between the current version and with I prefer `with`, as about creating lambdas, I don't see a problem with using them, but if you don't want to use them here ok.